### DISCUSSION TITLE
What's the purpose of this extension?

### DISCUSSION THREAD

**Discussion by tombohub** [microsoft/vscode-pylint#316] [Q&A] [answered]
I am confused, VScode pylint linting is already working without this extension:

![image](https://user-images.githubusercontent.com/3029412/230647973-acebd6d4-e297-4b61-b7ed-3e0f763c3fac.png)

so what exactly is this extension doing compared to already builtin pylint linting?

**Comment by karthiknadig** [comment, marked as answer]
This extension will replace the pylint support in python extension. This extension offers code actions for cases that can be fix by minor adjustment to code. It also makes it easier to contribute new fixes, and ships with pylint in the box so you don't have to install pylint in every environment.

You also get enhances problem description, with links to explanations:
![image](https://user-images.githubusercontent.com/3840081/230660306-e6bb11c6-c40d-4d9c-970c-0a9eecb19c94.png)

**Reply by tombohub** [reply]
thank you, so do I need to disable builtin pylint linting before I use this extension?

**Reply by karthiknadig** [reply]
You don't necessarily have to, but you will see two entries for the same error if you don't disable the python extension linting.

**Comment by ewerybody** [comment]
I'd add:
* If you don't need `pylint` installed in your project this extension will come with everything that's needed already!
* redundant options on the Python-extension will eventually be removed
